db/migrate/20160307120453_remove_hostgroups_count_from_puppetclasses.rb

```diff
@@ -0,0 +1,12 @@
+class RemoveHostgroupsCountFromPuppetclasses < ActiveRecord::Migration
+  def up
+    remove_column :puppetclasses, :hostgroups_count
+  end
+
+  def down
+    add_column :puppetclasses, :hostgroups_count, :integer, :default => 0
+    HostgroupClass.all.each do |hgc|
+      Puppetclass.increment_counter(:hostgroups_count, hgc.puppetclass_id)
```

Comment:
A query to group HostgroupClass by puppetclass_id and then update each puppetclass_id's counter?